op/model.py

```diff
@@ -91,6 +91,16 @@ def __init__(self, name, backend, cache):
     def __repr__(self):
         return f'<{type(self).__module__}.{type(self).__name__} {self.name}>'
 
+    @property
+    def is_leader(self):
+        if self.is_local:
+            # This value is not cached as it is not guaranteed to persist for the whole duration
+            # of a hook execution.
+            return self._backend.is_leader()
+        else:
+            # Unable to determine the leadership status for remote units.
+            raise NotLocalUnit(f"Unable to determine whether a remote unit {self} is a leader")
```

Comment:
This is a coding error which people should not need to catch, because they should know that they are iterating on other units vs. looking at the local unit. In this specific case it can be a RuntimeError.

If we do introduce something like LocalUnitError, we should also update every other similar case to emit the same error.